src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx

```diff
@@ -0,0 +1,257 @@
+
+const backend = defineBackend({
+  auth,
+  data,
+  myApiFunction,
+});
+
+// create a new API stack
+const apiStack = backend.createStack("api-stack");
+
+// create a IAM authorizer
+const iamAuthorizer = new HttpIamAuthorizer();
+
+// create a User Pool authorizer
+const userPoolAuthorizer = new HttpUserPoolAuthorizer(
+  "userPoolAuth",
+  backend.auth.resources.userPool,
+  {
+    userPoolClients: [backend.auth.resources.userPoolClient],
+  }
+);
+
+// create a new HTTP Lambda integration
+const httpLambdaIntegration = new HttpLambdaIntegration(
+  "LambdaIntegration",
+  backend.myApiFunction.resources.lambda
+);
+
+// create a new HTTP API with IAM as default authorizer
+const httpApi = new HttpApi(apiStack, "HttpApi", {
+  apiName: "myHttpApi",
+  defaultAuthorizer: iamAuthorizer,
+  corsPreflight: {
+    // Modify the CORS settings below to match your specific requirements
+    allowMethods: [
+      CorsHttpMethod.GET,
+      CorsHttpMethod.POST,
+      CorsHttpMethod.PUT,
+      CorsHttpMethod.DELETE,
+    ],
+    // Restrict this to domains you trust
+    allowOrigins: ["*"],
+    // Specify only the headers you need to allow
+    allowHeaders: ["*"],
+  },
+  createDefaultStage: true,
+});
+
+// add routes to the API with a IAM authorizer and different methods
+httpApi.addRoutes({
+  path: "/items",
+  methods: [HttpMethod.GET, HttpMethod.PUT, HttpMethod.POST, HttpMethod.DELETE],
+  integration: httpLambdaIntegration,
+  authorizer: iamAuthorizer,
+});
+
+// add a proxy resource path to the API
+httpApi.addRoutes({
+  path: "/items/{proxy+}",
+  methods: [HttpMethod.OPTIONS, HttpMethod.ANY],
+  integration: httpLambdaIntegration,
+});
+
+// add route to the API with a User Pool authorizer
+httpApi.addRoutes({
+  path: "/cognito-auth-path",
+  methods: [HttpMethod.GET],
+  integration: httpLambdaIntegration,
+  authorizer: userPoolAuthorizer,
+});
+
+// create a new IAM policy to allow Invoke access to the API
+const apiPolicy = new Policy(apiStack, "ApiPolicy", {
+  statements: [
+    new PolicyStatement({
+      actions: ["execute-api:Invoke"],
+      resources: [
+        `${httpApi.arnForExecuteApi("items")}`,
+        `${httpApi.arnForExecuteApi("cognito-auth-path")}`,
+      ],
+    }),
+  ],
+});
+
+// attach the policy to the authenticated and unauthenticated IAM roles
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(apiPolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(apiPolicy);
+
+// add outputs to the configuration file
+backend.addOutput({
+  custom: {
+    API: {
+      [httpApi.httpApiName!]: {
+        endpoint: httpApi.url,
+        region: Stack.of(httpApi).region,
+        apiName: httpApi.httpApiName,
+      },
+    },
+  },
+});
+```
+
+## Install Amplify Libraries
+
+<InlineFilter filters={['javascript', "angular", "react", "vue", "react-native", "nextjs"]}>
+Use the package manager of your choice to install the amplify JS library. For example, with `npm`:
```

Comment:
```suggestion
Use the package manager of your choice to install the Amplify JavaScript library. For example, with `npm`:
```

consolidating the reference to the js library (could also call this "client library" or "Amplify client library"